packages/cds-annotation-parser/src/transformer/annotationAstNodes.ts

```diff
@@ -0,0 +1,279 @@
+import { Range } from 'vscode-languageserver-types';
+import { copyRange } from './range';
+
+export type QualifiedName = string;
+export type SimpleIdentifier = string;
+export enum Delimiter {
+    none = 'none',
+    quoted = 'quoted',
+    exclamationSquareBrackets = 'exclamation-square-brackets'
+}
+export interface Node {
+    type: string;
+    range?: Range;
+}
+
+export const TOKEN_TYPE = 'token';
+export interface Token extends Node {
+    type: typeof TOKEN_TYPE;
+    value: string;
+}
+
+export const NUMBER_LITERAL_TYPE = 'number';
+export interface NumberLiteral extends Node {
+    type: typeof NUMBER_LITERAL_TYPE;
+    value: number | string;
+}
+
+export const ENUM_TYPE = 'enum';
+export interface Enum extends Node {
+    type: typeof ENUM_TYPE;
+    path: Path;
+}
+
+export type QuotedLiteralKind = 'date' | 'time' | 'timestamp' | 'binary';
+
+export const QUOTED_LITERAL_TYPE = 'quoted-literal';
+export interface QuotedLiteral extends Node {
+    type: typeof QUOTED_LITERAL_TYPE;
+    value: string;
+    kind: QuotedLiteralKind;
+}
+
+export const EMPTY_VALUE_TYPE = 'empty-value';
+export interface EmptyValue extends Node {
+    type: typeof EMPTY_VALUE_TYPE;
+}
+
+export const BOOLEAN_TYPE = 'boolean';
+export interface BooleanLiteral extends Node {
+    type: typeof BOOLEAN_TYPE;
+    value: boolean;
+}
+
+export const STRING_LITERAL_TYPE = 'string';
+export interface StringLiteral extends Node {
+    type: typeof STRING_LITERAL_TYPE;
+    value: string;
+    openToken?: Token;
+    closeToken?: Token;
+}
+export const MULTI_LINE_STRING_LITERAL_TYPE = 'multi-line-string';
+export interface MultiLineStringLiteral extends Node {
+    type: typeof MULTI_LINE_STRING_LITERAL_TYPE;
+    stripIndentation: boolean;
+    value: string;
+    openToken?: Token;
+    closeToken?: Token;
+}
+
+export const PATH_TYPE = 'path';
+export interface Path extends Node {
+    type: typeof PATH_TYPE;
+    value: string;
+    segments: Identifier[];
+    separators: Separator[];
+}
+
+export const SEPARATOR_TYPE = 'separator';
+export interface Separator extends Node {
+    type: typeof SEPARATOR_TYPE;
+    value: string;
+    escaped: boolean;
+}
+
+export const IDENTIFIER_TYPE = 'identifier';
+export interface Identifier extends Node {
+    type: typeof IDENTIFIER_TYPE;
+    value: string;
+    quoted?: boolean;
+}
+
+export const OPERATOR_TYPE = 'operator';
+export interface Operator extends Node {
+    type: typeof OPERATOR_TYPE;
+    value: string; // image of token representing operator (e.g. '-')
+    range: Range;
+}
+
+export type EXPRESSION_TYPE =
+    | typeof INCORRECT_EXPRESSION_TYPE
+    | typeof UNSUPPORTED_OPERATOR_EXPRESSION_TYPE
+    | typeof CORRECT_EXPRESSION_TYPE;
+export interface Expression extends Node {
```

Comment:
Please add the fix that was needed for CDS converter here already.
We should have `BaseExpression` interface without type and `Expression` type should be a union type with the 3 kinds of expressions.

Reply:
Done